GSA-TTS/FAC
# 3441 web search should properly disseminate tribal data

## Description
This PR is to:
- Allow tribal submissions to appear in search results for all users
- Allow PDF downloads to work for users with the read-tribal permission
- Remove PDF download(s) button for users without the read-tribal permission
- See #3441 
- For testing purposes: Submit two reports, then make one private by unchecking the is_public flag in the dissemination_general table. Confirm that you can view both reports through the search interface, regardless of whether you are logged in. Also, verify that you can access the summary page for both reports but are unable to use the download button for the PDF of the private report.
- The change related to summary workbook will be part of a separate PR

-----
In this PR (Phil's stuff):
- Summary workbook downloads are unabridged for users with the read-tribal permission and omit [all protected fields](https://docs.google.com/spreadsheets/d/1DXkWcu3kuAOeQh4B3YrdiiHnIIc04QdLyzJwDLyjneg/edit#gid=0) for users without it
- Disclaimers added to all tribal-related cover sheets

Testing:
- Submit two reports, then make one private by unchecking the is_public flag in the dissemination_general table
- As a non-privileged user:
  - Go to http://localhost:8000/dissemination/search/ and perform a search that includes both reports
  - Click the Download all button
  - The resulting spreadsheet should have the tribal report_id rows missing from the `notes`, `captext`, and `findingtext` sheets, but all other (public) rows remaining
  - View the tribal audit and click the SF-SAC download button. The same rows should be removed here as well.
     - The cover sheet disclaimer should be as described [here](https://github.com/GSA-TTS/FAC/issues/3441#issuecomment-1973504324)
- As a privileged user
  - Repeat the steps above. All rows (including tribal report_id ones) should be there.
  - View the tribal audit and click the SF-SAC download button. All rows should be here as well.
    - The cover sheet disclaimer should be as described [here](https://github.com/GSA-TTS/FAC/issues/3441#issuecomment-1973504324)
 
## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
